### ISSUE TITLE
Is it possible to perform side effects while parsing?

### ISSUE THREAD

**Issue by loewenheim** [softdevteam/grmtools#377]
I am wondering if it is possible, instead of parsing source text into an AST, to output a vector of bytes in postfix order. For example, assuming constants are prefixed by `17` (for no particular reason) and `+` is encoded as `5`, I would want `2 + 11` to be parsed into the vector `[17, 2, 17, 11, 5]`. This would require allocating the vector at the start of the parsing process and then pushing to it as expressions are parsed. I haven't seen anything in the documentation about how to do this, so I suspect it may not be supported.

Context: The bytecode interpreter in [Crafting Interpreters](https://craftinginterpreters.com/a-bytecode-virtual-machine.html) uses this kind of architecture. It's easy to do if you write a recursive descent parser as the book advocates, but I'm interested in using a parser generator.

**Comment by ltratt** [comment]
Honest answer: I don't remember. I think "no" but I know I looked at this at some point, and I don't remember what I settled upon.

That said, in general I'd say that in LR parsing it's better to bubble results up then force mutation downwards. Flattening things (as in your example) is a common idiom https://softdevteam.github.io/grmtools/latest_release/book/parsing_idioms.html#define-a-flatten-function -- would that work?
